File: 13.py
class Solution:
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        values = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        l = len(s)
        inv_s = s[::-1]
        res = values[inv_s[0]]
        for i in range(l-1):
            if values[inv_s[i]] > values[inv_s[i+1]]:
                res -= values[inv_s[i+1]]
            else:
                res += values[inv_s[i+1]]
        return res

File: 13.java
class Solution {
    public int romanToInt(String s) {
        char prev = ' ';
        int res = 0;
        for (char c: s.toCharArray()) {
            if (c == 'I') {
                res++;
            } else if (c == 'V') {
                res += 5;
                if (prev == 'I') res -=2;
            } else if (c == 'X') {
                res += 10;
                if (prev == 'I') res -=2;
            } else if (c == 'L') {
                res += 50;
                if (prev == 'X') res -= 20;
            } else if (c == 'C') {
                res += 100;
                if (prev == 'X') res -= 20;
            } else if (c == 'D') {
                res += 500;
                if (prev == 'C') res -= 200;
            } else {
                res += 1000;
                if (prev == 'C') res -= 200;
            }
            prev = c;
        }
        return res;
    }
}